compiler/compile/OMRCompilation.cpp

```diff
@@ -720,7 +720,22 @@ OMR::Compilation::getOSRInductionOffset(TR::Node *node)
 bool
 OMR::Compilation::requiresPreOSRPoint(TR::Node *node)
    {
-   return node->getOpCodeValue() != TR::monent;
+   // When there is no OSR induction offset, a pre OSR point is required
+   // This currently only covers async checks as they save the state when the check is performed
+   if (getOSRInductionOffset(node) == 0)
```

Comment:
Just curious why asyncchecks and monents are not in the same category as in neither requiring a pre OSR point ? Is'nt OSR really only possible once an asynccheck returns back to compiled code (same as a monent) ?